recipe/meta.yaml

```diff
@@ -1,34 +1,92 @@
 {% set name = "torchani" %}
-{% set version = "2.2" %}
+{% set version = "2.2.1" %}
+{% set torchani_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
+# Notes: this receipt is based on torchvision-feedstock:
+# https://github.com/conda-forge/torchvision-feedstock/blob/master/recipe/meta.yaml
+
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: c086fe6cabe49a0408d1f1b12aaf14e33b9e6ef0ffb15f3020792934e7c9fbbe
-  
+  url: https://github.com/aiqm/torchani/archive/refs/tags/{{ version }}.tar.gz
+  sha256: c4f299a586518c57b17332309f803e6ae34c1b7a4b48d542bd6c46e9c4f8d146
+
 build:
-  noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}                                                  # [cuda_compiler_version == "None"]
+  string: py{{ CONDA_PY }}cuda{{ cuda_compiler_version|replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}  # [cuda_compiler_version != "None"]
+  script_env:
+  - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
+  script:
+    - "{{ PYTHON }} -m pip install . --global-option='--cuaev-all-sms' -vvv --no-deps"  # [cuda_compiler_version != "None"]
+    - "{{ PYTHON }} -m pip install . -vvv --no-deps"                                    # [cuda_compiler_version == "None"]
 
 requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - sysroot_linux-64 ==2.17                # [linux64]
+
   host:
     - python
     - pip
     - setuptools_scm
+    - importlib_metadata
+    - numpy
+    - pytorch
+    - pytorch-gpu {{ pytorch_version }}  # [cuda_compiler_version != "None"]
+    - pytorch-cpu {{ pytorch_version }}  # [cuda_compiler_version == "None"]
+    # cuda 10.2 does not have cub, which need to be manually downloaded in setup.py
+    - wget                               # [cuda_compiler_version == "10.2"]
+    - unzip                              # [cuda_compiler_version == "10.2"]
```

Comment:
Can we try to use the [`cub`](https://github.com/conda-forge/cub-feedstock) from conda-forge for this? Downloading binaries in the build process is really something we should try to avoid.

Reply:
cub seems to be an header only library. But otherwise I agree with you.

Maybe a flag could be added to setup.py to prevent installing cub?

Reply:
See for the previous discussion about `cub`: https://github.com/conda-forge/torchani-feedstock/pull/2#discussion_r739841071

---

I confirm it's a header library only:

> CUB is implemented as a C++ header library. There is no need to build CUB separately. To use CUB primitives in your code, simply:

Search for "How do I get started using CUB?" at https://nvlabs.github.io/cub/.

I think it's fine to let it that way (also `cub` is only for cuda 10.2). If you have a strong opinion @h-vetinari, please let us know.

Reply:
cub is only needed for 10.2. And to be honest we would like to drop 10.2 asap when we could. 
So if possible we don't want to wasting time to modify our code to include conda-forge cub when pytorch compiling it.

Reply:
> If you have a strong opinion @h-vetinari, please let us know.

Fine for me to leave as is, just wanted to comment since it stood out to me.